clients/venice-push-job/src/main/java/com/linkedin/venice/hadoop/DefaultInputDataInfoProvider.java

```diff
@@ -365,35 +351,41 @@ private Pair<Schema, Schema> checkAvroSchemaConsistency(
   protected Pair<Schema, Schema> getAvroFileHeader(FileSystem fs, Path path, boolean isZstdDictCreationRequired) {
     LOGGER.debug("path:{}", path.toUri().getPath());
     VeniceAvroRecordReader recordReader = getVeniceAvroRecordReader(fs, path);
+    VeniceAvroFileIterator fileIterator = new VeniceAvroFileIterator(fs, path, recordReader);
 
     if (isZstdDictCreationRequired) {
       if (!pushJobSetting.isIncrementalPush) {
         if (!pushJobSetting.useMapperToBuildDict) {
           /** If dictionary compression is enabled for version, read the records to get training samples */
-          if (storeSetting.compressionStrategy == CompressionStrategy.ZSTD_WITH_DICT) {
-            InputDataInfoProvider.loadZstdTrainingSamples(recordReader, pushJobZstdConfig);
+          if (pushJobSetting.storeCompressionStrategy == CompressionStrategy.ZSTD_WITH_DICT) {
+            InputDataInfoProvider.loadZstdTrainingSamples(fileIterator, pushJobZstdConfig);
           }
         } else {
           /** If dictionary compression is enabled for version or compression metric collection is enabled,
            * read the records to get training samples
            */
-          InputDataInfoProvider.loadZstdTrainingSamples(recordReader, pushJobZstdConfig);
+          InputDataInfoProvider.loadZstdTrainingSamples(fileIterator, pushJobZstdConfig);
         }
       }
     }
-    return new Pair<>(recordReader.getFileSchema(), recordReader.getStoreSchema());
+
+    Utils.closeQuietlyWithErrorLogged(fileIterator);
```

Comment:
We could use try with resource?

Reply:
Yeah, that would be better. I didn't do that since I'd have to catch the exceptions and handle them, but it is a safer code style. Will update it